permissions_validators/src/public_blockchain/burn.rs

```diff
@@ -76,12 +85,7 @@ impl<W: WorldTrait> HasToken<W> for GrantedByAssetCreator {
             return Err("Destination is not an Asset.".to_owned());
         };
 
-        Ok(
-            PermissionToken::new(CAN_BURN_ASSET_WITH_DEFINITION.clone()).with_params([(
-                ASSET_DEFINITION_ID_TOKEN_PARAM_NAME.to_owned(),
-                asset_id.definition_id.into(),
-            )]),
-        )
+        Ok(CanBurnAssetWithDefinition::new(asset_id.definition_id).into())
```

Comment:
I think we can change the  `HasToken` trait definition to:
```rs
impl<W: WorldTrait> HasToken<W> for GrantedByAssetCreator {
    fn token(
        &self,
        _authority: &AccountId,
        instruction: &Instruction,
        wsv: &WorldStateView<W>,
) -> Result<impl Into<PermissionToken>, String> {
```

Reply:
this should be possible if `HasToken` always returns one and the same type of specialized permission token

Reply:
> this should be possible if HasToken always returns one and the same type of specialized permission token

It makes sense to me that it _should_, but I'll double-check

Reply:
Actually I don't know if there's value in doing that, since the whole point of specialized tokens is that they are separate from core, so should we change core to accommodate them?

Reply:
I think we should, in places where possible, prefer the use of specialized tokens. Maybe there is some technical reason why it's not possible in this place though

Reply:
specialized tokens are not available in core lib?